docs/features/techdocs/creating-and-publishing.md

```diff
@@ -66,7 +66,9 @@ Update your component's entity description by adding the following lines to its
 ```yaml
 metadata:
   annotations:
-    backstage.io/techdocs-ref: dir:./
+    backstage.io/techdocs-ref: url:https://github.com/org/repo
```

Comment:
Is there anywhere in the Backstage docs where this is outlined/explained?  From the outside, I immediately have questions like...

- Is this a ref? What is a ref? Why does it take this form?
- What are the different "types" and why do they exist?
- What do the different types do?
- Which one should I use and when (and why)?

Reply:
All great questions, Eric. The annotation is described here https://backstage.io/docs/features/software-catalog/well-known-annotations#backstageiotechdocs-ref and the only docs on URL Reader are here https://backstage.io/docs/features/techdocs/how-to-guides#how-to-use-url-reader-in-techdocs-prepare-step

But we need a detailed reference guide for how the TARGET in `url:TARGET` should look like for a combination of GitHub, GitLab, Bitbucket, etc. with/without branch on either root or a sub path.

I am not too sure about promoting the usage of `dir:` type since it is only used in TechDocs and seems unnecessary to maintain alongside `url:` which is used across Backstage and plugins.

I'll add few lines here about the annotation but will also cc @Rugvip for the "Integrations" section at backstage.io/docs where we can write the Spec for URL Reader.